generators/assets/imageGenerator.js

```diff
@@ -127,6 +127,62 @@ const generateResizedAssets = (
   );
 };
 
+const generateResizedRoundAssets = (
+  sourcePath,
+  destinationPath,
+  width,
+  givenHeight
+) => {
+  const height = givenHeight || width;
+
+  const directory = path.dirname(destinationPath);
+  if (!fs.existsSync(directory)) {
+    fs.mkdirpSync(directory);
+  }
+
+  const psdSafeSourcePath = `${sourcePath}${
+    sourcePath.split('.').pop() === 'psd' ? '[0]' : ''
+  }`;
+  const maxSize = Math.max(width, height);
+  // percent margin
+  const marginPercent = 90 / 100;
+
+  // create mask
+  gm(maxSize, maxSize, 'none')
+    .drawCircle(
+      maxSize / 2,
+      maxSize / 2,
+      maxSize / 2,
+      (maxSize * (1 - marginPercent)) / 2
+    )
+    .gravity('Center')
+    .writeAsync(path.normalize(destinationPath + '.mask.png'))
+    .then(() => {
+      console.log(`Wrote ${destinationPath + '.mask.png'}`);
+    });
+
+  gm(path.normalize(psdSafeSourcePath))
```

Comment:
Composing of images depends on creating *.mask.png. So i think this gm chain for compose original png with *.mask.png should be places inside of  `.writeAsync` `then(() => {` when writeAsync is asynchronous.